parasoft-findings-teamcity-agent/src/main/resources/com/parasoft/findings/teamcity/agent/sa-pmdcpd.xsl

```diff
@@ -1,8 +1,52 @@
 <?xml version="1.0"?>
-<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:exsl="http://exslt.org/common">
     <xsl:output method="xml" encoding="UTF-8" indent="yes"/>
 
     <xsl:variable name="isStaticAnalysisResult" select="count(/ResultsSession/CodingStandards) = 1" />
+    <xsl:variable name="isCPPProReport" select="not(/ResultsSession/@prjModule) and /ResultsSession/@toolName = 'C++test'"/>
+
+    <!-- Handle file location for C/C++Test pro report -->
+    <xsl:variable name="locationMap">
+        <xsl:if test="$isCPPProReport">
+            <map>
```

Comment:
Since we use Java internal API(javax.xml.transform.Transformer) to tranform report, this API is in JDK. It seems JDK11 doesn't support XSLT 3.0, so we can not use many new feature in XSLT 3.0 like [xsl:map](https://www.saxonica.com/html/documentation12/xsl-elements/map.html) and [xsl:map-entry](https://www.saxonica.com/html/documentation12/xsl-elements/map-entry.html).
Here is the link for [Creating a Map in XSLT](https://9to5answer.com/xslt-creating-a-map-in-xslt) in XSLT 1.0.
I use the XSLT 1.0 -> Solution 2 method.

Reply:
> It seems JDK11 doesn't support XSLT 3.0

XPath/XSLT in Java is only 1.0.  You have to use Saxon if you want 2.0 or later.

Reply:
Thanks for your suggestion Benken. I need to discuss with Yuqian and decide to use Saxon or not this time.